scipy/stats/_continuous_distns.py

```diff
@@ -8854,6 +8854,149 @@ def _updated_ctor_param(self):
         return dct
 
 
+class studentized_range_gen(rv_continuous):
+    """
+    The generalized studentized range continuous random variable.
+
+    %(before_notes)s
+
+    See Also
+    --------
+    t: student's t distribution
+
+    Notes
+    -----
+    When `v` is arbitrarily large, in this implementation greater than or equal
+    to 2000, an asymptotic method is used in calculation of the cumulative
+    distribution function. [4]_
+
+    Integrals for the PDF, CDF, and moment are evaluated using
+    `scipy.integrate.nquad` with LowLevelCallables written using Cython.
+    Most of the integrals are evaluated using logarithms to prevent overflows
+    from the multiple exponential terms within. Furthermore, all terms are
+    evaluated within the integral to correctly control error.
+
+    The distribution's accuracy has been verified against R, statsmodels,
+    tables, with an mpmath implementation acting as ground truth.
+
+    References
+    ----------
+
+    .. [1] "Studentized range distribution",
+           https://en.wikipedia.org/wiki/Studentized_range_distribution
+    .. [2] Batista, Ben Dêivide, et al. "Externally Studentized Normal Midrange
+           Distribution." Ciência e Agrotecnologia, vol. 41, no. 4, 2017, pp.
+           378–389., doi:10.1590/1413-70542017414047716.
+    .. [3] Harter, H. Leon. “Tables of Range and Studentized Range.” The Annals
+           of Mathematical Statistics, vol. 31, no. 4, 1960, pp. 1122–1147.
+           JSTOR, www.jstor.org/stable/2237810. Accessed 18 Feb. 2021.
+    .. [4] Lund, R. E., and J. R. Lund. “Algorithm AS 190: Probabilities and
+           Upper Quantiles for the Studentized Range.” Journal of the Royal
+           Statistical Society. Series C (Applied Statistics), vol. 32, no. 2,
+           1983, pp. 204–210. JSTOR, www.jstor.org/stable/2347300. Accessed 18
+           Feb. 2021.
+
+
+    %(example)s
```

Comment:
@mdhaber While looking into adding an example with what was done in gh-13431, I ran into a little issue with the inversion of the cdf. The range of `ks` is based on the support with the specified shapes, but the result is from 0 to infinity, which is troublesome to create a range for. 

```
from scipy.stats import studentized_range as srd
k, v = 3, 10
a, b = srd.support(k, v)
a, b
```
Out[113]: (0, inf)

It may still be possible to generate using just an arbitrarily large number, and then adding infinity onto it. Using np.geomspace would help there be more small-ish samples (There are more `small` q crit values than large, looking at the tables).

I'm not sure if there is a way to evaluate whether this holds up to publishing standards of rvs approximation, but it seems to work experimentally. I'm not sure what the best method to verify this would be -- it doesn't look like the log-likelihood function (from studentized_range._reduce_func...) really works to check data against parameters, results always appear to be infinity. 

So, it might be best to leave it how it is. Here's what I was working with in case you had other thoughts. 

```
import numpy as np
import scipy.stats as stats
from scipy.stats import studentized_range as srd
from scipy.interpolate import interp1d

random_state = np.random.default_rng(123)
size=100
k, v = 3, 10
a, b = srd.support(k, v)

ks = np.geomspace(1, 1200)
ks[-1] = np.inf
# ks = np.arange(a, 1200, dtype=np.float64)
cdf = srd.cdf(ks, k, v)
ppf = interp1d(cdf, ks, kind='next', fill_value='extrapolate')

rvs = ppf(random_state.uniform(size=size))
```

Reply:
Rather than give away the answer : )
* What is the largest possible random number generated by `rand`?
* Considering how inverse transform sampling works, how can this be used to pick a finite number instead of infinity for the upper end?

Reply:
- it looks like the the range for rand is the open interval of [0, 1]
- Inverse transform sampling uses the inversion of the CDF, the PPF, with uniform input x~[0,1]. So if 1 results in infinity in the arithmatically computer ppf,  picking a value ever so slightly smaller than 1 could get a finite bound. `srd.ppf(1, k, v) -> inf`.
```
In [16]: srd.ppf(1 * (1-np.finfo(float).eps), k, v)
Out[16]: 132.20863262903356
```

Reply:
Yup! That's what I had in mind.

Reply:
I made histograms with numbers of points in range(100, 1000, 50), and it starts to look good with the sample size around 350, taking ~4s. 

![image](https://user-images.githubusercontent.com/44255917/109097910-22452300-76d5-11eb-9e46-0cb26050617a.png)

Reply:
Yeah I think we're wasting a lot of resolution at the extreme right tail. `srd.ppf(1 * (1-np.finfo(float).eps), k, v)` is correct if you're going for machine precision, because it's impossible to generate a random number closer to one than that, but in our case, there's a small change we'll get a number more than, say, 0.999, and even if we do, there are so few of them that it doesn't matter what their value is; they won't affect the shape of the histogram where it matters. Try `srd.ppf(0.999, k, v)` as the upper end and I bet you can get by with a lot fewer points.

Reply:
You're right -- srd.ppf(0.999, k, v) gives  7.41. 

The histogram looks good even with 25 samples, so I think I'll settle it at 50. Then it takes only 400ms.

Reply:
Preview of the docs so you don't have to regenerate them https://pedantic-kirch-95b16a.netlify.app/scipy.stats.studentized_range%20%E2%80%94%20scipy%20v1.6.0.dev0+5d5f100%20reference%20guide